tests/Fixer/Operator/TernaryToNullCoalescingFixerTest.php

```diff
@@ -41,6 +41,8 @@ public static function provideFixCases(): iterable
             ['<?php $x = isset($a) and $a ? $a : "";'],
             ['<?php $x = "isset($a) ? $a : null";'],
             ['<?php $x = isset($a) ? $$a : null;'],
+            ['<?php $x = isset($this) ? $this : null;'],
```

Comment:
Let's wrap this case and all the others using `$this` (in any casing) with `class Foo { function bar() { /* ... */ }}`, so anyone seeing it in he future will know the context.

Reply:
But the whole point of it is to be outside of the class 🤔. Should these cases be cloned and wrapped, or just wrapped (replaced)?

Reply:
This can be present in an application, but the last one (which is `<?php $x = isset($this["foo"]) ? $this["foo"] : null;` will always result in a fatal error, so why even bother to change it? When inside of a class it is possible to have it in an application.

Reply:
Agree to remove those cases

Reply:
I agree that with current PHP behavior we can remove those cases. But personally [I consider it as a bug](https://twitter.com/_Codito_/status/1669080477452455939) 😉.

Reply:
Yeah, my case is that `Carbon` supports macro/mixin the Laravel way and you can call those macro statically (`Carbon::myMacro()`) or dynamically (`(new Carbon())->myMacro()`), so inside the definition of myMacro we want to make sure `$this` exists, so we bind a context to the closure.

Of course my first choice was to write (simplified version):
```php
$closure->bindTo($this ?? new Carbon());
```
Sadly triggers the exception, but then I discovered expending with `isset($this) ? $this :` works 🤷 then that's where cs-fixer suggested the change, but it was not my first choice.

It sure is inconsistent: null-coalescing could just be always equivalent to expended form.

Reply:
There a difference between [`isset()`](https://www.php.net/manual/en/function.isset.php) and null coallesce.

> Note: Because this is a language construct and not a function, it cannot be called using [variable functions](https://www.php.net/manual/en/functions.variable-functions.php), or [named arguments](https://www.php.net/manual/en/functions.arguments.php#functions.named-arguments).

Because of that, `isset()` can handle non-existing variables or even whole fetch chains silently. On the other hand `??` is an operator, so you need valid (existing) left side. What I personally consider as a bug is that [`isset($notExisting->some->thing)`](https://3v4l.org/b3qUC) works, while `isset($this->some->thing)` does NOT work, even though `isset($this)` alone works.